npe2/io_utils.py

```diff
@@ -162,6 +162,12 @@ def _read(
             # if the reader function raises an exception here, we don't try to catch it
             if layer_data := read_func(paths, stack=stack):
                 return (layer_data, rdr) if return_reader else layer_data
+
+    if plugin_name:
+        raise ValueError(
+            f"Plugin {plugin_name} was selected to open "
```

Comment:
might also make this `{plugin_name!r}` ... and then change the test match to `Plugin 'nope' was selected`?